src/utils/version.ts

```diff
@@ -104,7 +104,7 @@ export function versionGreaterOrEqualThan(v1: SemVer, v2: SemVer): boolean {
 /**
  * A regular expression to detect that a version is a pre-release version.
  */
-export const PREVIEW_RELEASE_REGEX = /(?:[^a-z])(preview|pre|rc|dev|alpha|beta|unstable|a|b)(?:[^a-z]|$)/i;
+export const PREVIEW_RELEASE_REGEX = /(?:[^a-z])(preview|pre|rc|dev|alpha|beta|unstable|a|b|dump)(?:[^a-z]|$)/i;
```

Comment:
I've never heard of "dump" as a prerelease naming scheme -- is this standardized anywhere? I'd prefer we use actual semver names rather than making up our own here (as that would be more expected from users as well when reasoning about versions)

Reply:
As far as I understand, `dump` was chosen because it described a feature that was added to this preview release. So likely not standardized. I agree that this is a niche/edge case but "dump" kinda seems likely enough to come up again. Also noticed while [removing the entries](https://github.com/getsentry/sentry-release-registry/pull/169) of this version from the registry, we had similar naming schemes being used in the past.

I can also close this as I get your point. WDYT about checking for any suffix after the semver patch number instead?

Reply:
ok then we definitely shouldn't add this to the regex

I think craft should *reject* any non-semver versions -- I thought it already did this but maybe that's only for python?

the dotnet sdk release should have named it -pre1 or one of the standard naming schemes

Reply:
one important case either way is `1.2.3-1` is standardized [in python](https://packaging.python.org/en/latest/specifications/version-specifiers/#version-specifiers) as equivalent to `1.2.3.post1` and [we use that intentionally](https://github.com/getsentry/sentry-forked-django-stubs/releases) so we have to make sure that still works

Reply:
I checked the `1.2.3-1` case:

- `isPreviewRelease` returns `false`. I guess this is correct, given this is a post release? 
- `parseVersion` can parse this string and identifies the `-1` as `.pre`:
  ```js
   test('parses a Python-style post release version', () => {
     expect(parseVersion('1.2.3-1')).toEqual({
       major: 1,
       minor: 2,
       patch: 3,
       pre: '1',
     });
   });
   ```

Reply:
seems fine to me!

Reply:
Ok now the specific list of pre-releases makes sense: If we want to ensure Python post releases are not identified as pre-releases, we can't check against arbitrary pre-release version identifiers. So just checking for `!!parseVersion(...).pre` would not cover the python post release correctly. 

So I guess, we don't change anything 👍 I can open a PR with some tests I wrote to have the Python case covered.